archive/100-theorems-list/37_solution_of_cubic.lean

```diff
@@ -0,0 +1,199 @@
+/-
+Copyright (c) 2021 Jeoff Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeoff Lee
+-/
+import tactic.basic
+import data.complex.basic
+import algebra.group.defs
+
+/-!
+# The Solution of Cubic
+
+This file proves Theorem 37 from the [100 Theorems List](https://www.cs.ru.nl/~freek/100/).
+
+The theorem `cubic_eq_zero_iff` gives solution to the cubic equation over ℂ,
+`cubic_eq_zero_iff_of_discrim_eq_zero`
+based on the [Cardano's Formula](https://en.wikipedia.org/wiki/Cubic_equation#Cardano's_formula).
+
+## References
+
+Originally ported from Isabelle/HOL. The
+[original file](https://isabelle.in.tum.de/dist/library/HOL/HOL-ex/Cubic_Quartic.html) was written by Amine Chaieb.
+
+## Tags
+polynomial, cubic, root
+-/
+
+section
+
+open complex
+
+variables (a b c d : ℂ)
+variables {p q r s t : ℂ}
+
+noncomputable def ω : ℂ := ⟨-1/2, (real.sqrt 3)/2⟩
+
+lemma omega_cube_eq_one : ω ^ 3 = 1 :=
+begin
+  unfold ω,
+  have h3 : real.sqrt 3 * real.sqrt 3 = 3,
+  { rw [← pow_two (real.sqrt 3), @real.sq_sqrt 3 (by linarith)], },
+  rw [pow_succ, pow_succ, pow_one],
+  ext,
+  iterate 2
+  { simp only [mul_re, mul_im],
+    rw ← sub_eq_zero,
+    field_simp [mul_add, h3],
+    ring }
+end
+
+lemma omega_sum : 1 + ω + ω^2 = 0 :=
```

Comment:
available as `polynomial.cyclotomic_eq_geom_sum` and `polynomial.is_root_cyclotomic` I think.